src/yaml/serialization.cr

```diff
@@ -404,6 +404,11 @@ module YAML
             {% if key.is_a?(Path) %}
               {% key = key.resolve %}
             {% end %}
+            {% if key.is_a?(NumberLiteral) %}
+              # An enum value is always a typed NumberLiteral and stringifies with type
+              # suffix unless it's Int32.
+              {% key = key.id.split("_")[0] %}
```

Comment:
We should be sure to add a spec for `_` based numbers.  E.g. `100_000_u32`.

Reply:
Already in the patch.